cve_bin_tool/version_compare.py

```diff
@@ -140,10 +140,14 @@ def version_compare(v1: str, v2: str):
                 # Honestly it's hard to guess if .dev3 is going to be more or less than .rc4
                 # unless you know the project, so hopefully people don't expect that kind of range
                 # matching
```

Comment:
```suggestion
                # We will be dictionary comparing so that 4.alpha4 < 4.beta1 
                # but this also means .dev3 < .rc4 (because d is before r) which may be less logical.
```

I think this comment might be more helpful: explaining what happens and when that works/doesn't.